modules/dxp/apps/saml/saml-web/src/main/resources/META-INF/resources/admin/user_attribute_mapping.jsp

```diff
@@ -133,26 +136,23 @@ String userIdentifierExpression = attributeMappingDisplayContext.getUserIdentifi
 			.querySelectorAll(
 				'input[name="<portlet:namespace />attribute:userIdentifierExpressionIndex"]'
 			)
-			.forEach((radioControl) => (radioControl.checked = false));
+			.forEach((radioControl) => {
+				radioControl.checked = false;
+				radioControl.disabled = true;
+				radioControl.closest('label').classList.toggle('disabled', true);
+			});
 	};
 
-	<portlet:namespace />handleAttributeMappingMatchingSelection = function (row) {
+	<portlet:namespace />handleAttributeMappingMatchingSelection = function (row, attributeMappingSelected) {
 		document.querySelector(
 			'input[name="<portlet:namespace />attribute:userIdentifierExpressionPrefix"]'
 		).value = row.dataset.prefix;
 
-		document
-			.querySelectorAll(
-				'input[name="<portlet:namespace />userIdentifierExpression"]'
-			)
-			.forEach(
-				(userIdentifierExpressionRadioControl) =>
-					(userIdentifierExpressionRadioControl.checked = false)
-			);
-
-		document.querySelector(
-			'input[name="<portlet:namespace />userIdentifierExpression"][value="attribute"]'
-		).checked = true;
+		if (!attributeMappingSelected) {
+			document.querySelector('input[name="<portlet:namespace />userIdentifierExpression"][value="attribute"]').checked = true;
+			document.querySelectorAll('.user-attribute-mapping-row').forEach(
+				(row) => <portlet:namespace />evaluateAttributeMappingRow(row));
+		}
```

Comment:
It took me a while to appreciate why you had to introduce this new `attributeMappingSelected` parameter and include special handling when it is false. It is purely initialization code right? I have sent some commits to extract this into one time executed code, and hopefully the code that remains is then more succinct.

It also avoids this nested loop that I believe you introduced because already iterated rows need to be re-processed because of the mid-iteration change to the underlying form field's value.

Reply:
Yes, the `attributeMappingSelected` variable was required to enable the radio buttons when the user opens an IDP for editing. Without that the radio button was selected but disabled because we did not save the User Resolution Option only which field is used for matching, so when we load the page I checked if any field is selected for matching, selected the required User Resolution Option and iterated through all rows again to enable them.

Reply:
Exactly. The explicit selection of a field for matching is a bit of an odd one out because this selection is a parameter to the User Resolution Option, and we have to encode that into a single field storage in a format like 
`attribute:{userFieldExpression}`. For example `attribute:expando:integerField` . Normally it would just be stored as `dynamic` or `none`.